### PR TITLE
ci: Add workflow check for Merge Queue pipeline start on Azure Pipeline 

### PR DESCRIPTION
**Pull request by jpayne3506**
**Reason for Change**:

Adds a workflow check to ensure that merge queues get a pipeline run assigned to them if the CI trigger fails. 

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

File: .github/workflows/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+name: Azure Pipeline Merge Queue Check
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+    - master
```
Comment (vipul-21) [inline .github/workflows/pipeline.yaml:6]: should this added to `release` branches too ?

File: .github/workflows/pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      env:
+        AZURE_CORE_OUTPUT: none
+      with:
+        azcliversion: latest
+        inlineScript: |
+          az account show
+          echo ${{ secrets.AZURE_DEVOPS_EXT_PAT }} | az devops login --org ${{ secrets.AZURE_PIPELINE_ORG }}
+
+          echo "Sanity check recent Pipeline runs"
+          az pipelines runs list --pipeline-ids ${{ secrets.AZURE_PIPELINE_ID }} --org ${{ secrets.AZURE_PIPELINE_ORG }} --project ${{ secrets.AZURE_PIPELINE_PROJECT }} --reason individualCI --top 10 --output json | jq -r .[].sourceBranch
```
Comment (vipul-21) [inline .github/workflows/pipeline.yaml:41]: What's the use of the call ?

File: .github/workflows/pipeline.yaml (hunk 1 above)
Comment (jpayne3506) [reply .github/workflows/pipeline.yaml:6]: We don't need to add this to release/* branches as they do not use merge queue to merge.

File: .github/workflows/pipeline.yaml (hunk 2 above)
Comment (jpayne3506) [reply .github/workflows/pipeline.yaml:41]: On the happy path where AZP and GH properly queue up a pipeline run before this workflow start, line 41's output will show the merge queue PR branch i.e. `refs/heads/gh-readonly-queue/master/pr-2460-****` as part of the 10 most recent `individualCI` runs. 

We can check `all, batchedCI, buildCompletion, checkInShelveset, individualCI, manual, pullRequest, schedule, triggered, userCreated, validateShelveset` for the reason of the queued run, but the only ones we deal with are `individualCI, manual, pullRequest`. 

Of note, how these runs will be triggered:
Happy path Merge Queue - individualCI
Workflow Check Run using az pipelines (line 60) - manual
Precheck from PR - pullRequest

This line is purely for debugging and adds additional context for the current status of pipeline.

**Review by vipul-21 — APPROVED** [review]

**Review by rbtr — APPROVED** [review]

**Pull request merged**